Added RPCs for the API Key mechanism

dylanratcliffe opened the pull request:
Here's my first stab at the API Key RPCs. Let me know what you think. The callback URL isn't included as the request won't be able to be parsed as a native Connect request, so we'll have to handle that manually. Shouldn't be too hard though

DavidS-om commented on apikey.proto:
```diff
@@ -0,0 +1,167 @@
+// API Calls //
+///////////////
+
+message ExchangeKeyForTokenRequest {
+    // The API Key that you want to exchange for an Oauth access token
+    string apiKey = 1;
+}
+
+message ExchangeKeyForTokenResponse {
+    // The access token that can now be use to authenticate to Overmind and its
+    // APIs
+    string accessToken = 1;
+}
+
```
Move this to the bottom, to preserve the order of RPCs

```suggestion
```

DavidS-om commented on apikey.proto:
```diff
@@ -0,0 +1,167 @@
+    bytes uuid = 1;
+}
+
+message DeleteAPIKeyResponse {}
```
```suggestion
message DeleteAPIKeyResponse {}

message ExchangeKeyForTokenRequest {
    // The API Key that you want to exchange for an Oauth access token
    string apiKey = 1;
}

message ExchangeKeyForTokenResponse {
    // The access token that can now be use to authenticate to Overmind and its
    // APIs
    string accessToken = 1;
}
```

DavidS-om commented on apikey.proto:
```diff
@@ -0,0 +1,167 @@
+
+    // The URL that the user should visit in order to authorize the newly
+    // created key. This will allow Auth0 to generate the access and refresh
+    // tokens and pass them to the API server. The user will be redirected once
```
```suggestion
    // tokens and pass them to the API server. The user will be redirected back to the frontend once
```

??

DavidS-om commented on apikey.proto:
```diff
@@ -0,0 +1,167 @@
+
+    // The scopes that the key should have
+    repeated string scopes = 2;
+
+    // The URL that the user should be redirected to after authorizing the key
+    string redirectURL = 3;
```
```suggestion
```

The redirectUrl is a deployment-specific implementation detail of the ApiKey service and doesn't need to (can't!) be supplied by an external client, therefore it doesn't need to be supplied here.

DavidS-om reviewed (changes requested):
Besides a few editorial changes, one big one re the redirect URL: Since that has to be configured in the Auth0 application config, we can't specify it here. It'll have to be something hardcoded (from the deployment config) where we correlate the incoming request through the attached `state`. Then we can forward the user back to the frontend.

Writing this I realise that that was maybe the intention, but that doesn't come through in the comments and re-using the OAuth terminology doesn't help either. In that case it should be something more specific like `frontendLandingPage`.

dylanratcliffe commented:
You're absolutely right. Good catch with the `state` param. I've updated the comments to reflect the static URL end use of the `state` param to determine the UUID, and also renamed the `redirectURL` to `finalFrontendRedirect` to distinguish it from OAuth stuff
